src/metabase/sync/util.clj

```diff
@@ -304,7 +304,7 @@
 
 (defmethod name-for-logging Database
   [{database-name :name, id :id, engine :engine,}]
-  (trs "{0} Database {1} ''{2}''" (name engine) (or id "") database-name))
+  (trs "{0} Database {1} ''{2}''" (name engine) (str (or id "")) database-name))
```

Comment:
when `id` is a number, trs will apply locale settings. Therefore id `13371337` -> "13,371,337"; running it through `str` skips that.